app/models/conditions_response/h_branding_fee_due_alert.rb

```diff
@@ -0,0 +1,34 @@
+# This emails all current members in a company if the h-branding fee is past due.
+#
+#
+class HBrandingFeeDueAlert < CompanyEmailAlert
+
+
+  # If an H Branding Fee is due for the company, then
+  #   send the alert if today is in the configuration list of days
+  #
+  def send_alert_this_day?(timing, config, company)
+
+    if RequirementsForHBrandingFeeDue.requirements_met?({company: company})
+
+      day_to_check = self.class.days_today_is_away_from(company.next_hbranding_payment_due_date, timing)
```

Comment:
Doesn't this need to change to use the expiration date as the "pivot date" for calculations?  That is, calculate days from the _expiration date_, instead of the day after?

Reply:
It _does_ base the calculation on the expiration date (assuming there has been a previous payment, of course.  See "situation 1" above https://github.com/AgileVentures/shf-project/pull/600#discussion_r247793510)

But a payment isn't due until the next day.  That's the whole reason for this feature.

Reply:
Maybe I'm reading this incorrectly, but - for the case with a prior, expired payment - `next_hbranding_payment_due_date` will return the day _after_ expiration.  If our concept is to use "days from _expiration_ date then we're one days off.

So, if I have a condition with days: [1,7, ....] I won't send an alert until the _second_ day after expiration, when my intent is to send an alert the day after.

Reply:
And in the case where there is no previous payment?  If the `days:` are based on expiration date (even when there isn't one -- which is a weirdness), then we will have to ensure that we never enter the condition with 0 in the `days:` array because that would be totally incorrect for this feature.  That seems like we're letting the implementation drive how we're accomplishing this, and how users (the admin) will have to interact with this.  

This feature is about the _due date._  And that's what should be reflected in the `days:` that the alert goes out.  

Definitely another good discussion needs to be had. :-)

Reply:
This is only a _thing_ because you raise such good points, @patmbolger .  If you weren't so intelligent, this would have been quicker. ;-)